src/rpc/net.cpp

```diff
@@ -114,7 +114,7 @@ static RPCHelpMan getpeerinfo()
                     {
                         {RPCResult::Type::STR, "SERVICE_NAME", "the service name if it is recognised"}
                     }},
-                    {RPCResult::Type::BOOL, "relaytxes", /*optional=*/true, "Whether we relay transactions to this peer"},
+                    {RPCResult::Type::BOOL, "relaytxes", /*optional=*/true, "Whether we relay transactions to this peer (not available while peer connection is being set up)"},
```

Comment:
As soon as getpeerinfo reports a new peer, we also have CNodeStateStats entry - but no TxRelay object yet. This means, we report `"relaytxes": false` in getpeerinfo 
([code](https://github.com/jonatack/bitcoin/blob/0e4b959e0a3874233b1549f9e7a257f47d40539f/src/net_processing.cpp#L1578)) when the connection is being set up, but the field should be available.

Reply:
Agree, it was occurring at disconnection rather than at connection.